### DISCUSSION TITLE
How can I pay in float instead of integer

### DISCUSSION THREAD

**Discussion by amit-eps** [bavix/laravel-wallet#927] [Q&A] [answered]
I'm using version "bavix/laravel-wallet": "^7.3"
How can I pay in float instead of integer

I'm using code like this to purchase product

 ```
  public function getAmountProduct(Customer $customer): float
    {
        return 0.5;
    }

  $user->pay($product);
```

**Comment by rez1dent3** [comment, marked as answer]
Hello. The package architecture is built around integer. Non-float methods are called inside payments. This means that in this method you must specify the cost in int depending on your business processes. For example, if you are sure that all wallets will have decimal places =2, then you can multiply the amount by 100.

Example:
```php
public function getAmountProduct(Customer $customer): float
{
//    return 5; // 0.05
//    return 50; // 0.5 
//    return 500; // 5  
    return 555; // 5.55 
}
```

**Reply by rez1dent3** [reply]
In version 11.x, a helper service appeared, which internally multiplies a number by 10^decimal_places. https://github.com/bavix/laravel-wallet/issues/831#issuecomment-1878336066
